app/views/user_sessions/_greeting.html.erb

```diff
@@ -6,24 +6,12 @@
         <%= link_to ts("Hi, %{current_user}!", :current_user => current_user.login, key: 'header'), user_path(current_user) %>
         <ul class="menu" role="menu">
           <li><%= span_if_current ts('My Dashboard', key: 'header'), user_path(current_user) %></li>
-          <% if @current_user.subscriptions.count > 0 %>
-            <li><%= span_if_current ts('My Subscriptions', key: 'header'), user_subscriptions_path(current_user) %></li>
-          <% end %>
-          <% if @current_user.visible_work_count > 0 %>
-            <li><%= span_if_current ts("My Works (%{works_number})", key: 'header', :works_number => @current_user.visible_work_count), user_works_path(@current_user) %></li>
-          <% end %>
-          <% if @current_user.bookmarks.count > 0 %>
-            <li><%= span_if_current ts("My Bookmarks (%{bookmark_number})", key: 'header', :bookmark_number => @current_user.bookmarks.count), user_bookmarks_path(current_user) %></li>
-          <% end %>
-          <% if @current_user.owned_collections.count > 0 %>
-            <li><%= span_if_current ts("My Collections (%{coll_number})", key: 'header', :coll_number => @current_user.owned_collections.count), user_collections_path(current_user) %></li>
-          <% end %>
-          <% if @current_user.challenge_signups.count > 0 %>
-            <li><%= span_if_current ts("My Sign-ups (%{signup_number})", key: 'header', :signup_number => @current_user.challenge_signups.count), user_signups_path(current_user) %></li>
-          <% end %>
-          <% if (@current_user.offer_assignments.undefaulted.count || @current_user.pinch_hit_assignments.undefaulted.count) > 0 %>
-            <li><%= span_if_current ts("My Assignments (%{assignment_number})", key: 'header', :assignment_number => (@current_user.offer_assignments.undefaulted.count + @current_user.pinch_hit_assignments.undefaulted.count)), user_assignments_path(current_user) %></li>
-          <% end %>
+          <li><%= span_if_current ts('My Subscriptions', key: 'header'), user_subscriptions_path(current_user) %></li>
```

Comment:
Can you please revive the discussion on the mailing list and explain what's going on here? None of us knew we would be agreeing to change how the menu works entirely, and I think our answers may change in light of this information. (I know personally, I would be okay with a compromise where we cache the counts for the conditional but didn't show the numbers in the link text. I wouldn't mind "My Works" lingering for a few minutes after I deleted my last work as much as I'd mind the number being wrong.)